src/extension/request_abort.c

```diff
@@ -167,6 +168,44 @@ void dd_set_response_code_and_type(int code, dd_response_type type)
     }
 }
 
+// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+void dd_set_redirect_code_and_location(int code, zend_string *nullable location)
+{
+    _response_code = code;
+    _redirection_location = location;
+}
+
+void dd_request_abort_redirect()
+{
+    if (!_redirection_location || ZSTR_LEN(_redirection_location) == 0) {
+        mlog(dd_log_warning, "Failing to redirect: No location set");
+        return;
+    }
```

Comment:
I'm not sure this is the best solution, we should still block / redirect even if there is some unexpected situation here. Perhaps we can just block with the default parameters?